hashicorp/terraform-provider-tfe
# Formatting the provider to use terraform-plugin-docs

## Description

Formatting provider to transition from manual markdown editing to using [tfplugindocs](https://github.com/hashicorp/terraform-plugin-docs) for [registry doc ](https://registry.terraform.io/providers/hashicorp/tfe/latest/docs) generation.

The followup PR #603 contains a guide script to help contributors prepare the necessary assets for creating new docs and run the `tfplugindocs` command.

## More Info
- My [lightning talk recording](https://hashicorp.zoom.us/rec/play/KD7gVOSnLvh2aHcdH38KhbTOx09NkZKPQ4JNzpBtz6xnBnybyPFR5z2BbcPwanSRJBnIWZTZgyKfGxmQ.TpftK41g1KmxuO3s?startTime=1659714349000) (3:00-10:00) has quick overview of what is tfplugindocs and the goal of this project.
- Directory "info-project-docsparkle" on the "docsparkle/scripts" also contains more info on the tool and details to be aware of.

### Summary of Changes (listed by directory)
**1. tfe**
  - resource/datasource descriptions defined in resource schema
  - attribute descriptions defined in resource schema

**2. examples**
  - code snippets for "example usage" section organized in specific subdirectories

**3. templates**
  - `.md.tmpl` used to generate markdown files
  - general fallback templates exist for resources and data-sources
  - custom templates exist for any that require additional sections beyond fallback

**4. docs**
  - automatically generated markdown after running `tfplugindocs` command
  - should never be directly edited

## Testing plan

1. install [tfplugindocs](https://github.com/hashicorp/terraform-plugin-docs#installation)
2. run `tfplugindocs` command
3. check the `docs` directory for generated `.md`
4. paste markdown in [docs preview tool](https://registry.terraform.io/tools/doc-preview) to see how it will appear on the website
